### PR TITLE
Introduce InstanceSym

### PR DESCRIPTION
**Pull request by mlutze**
Fixes https://github.com/flix/flix/issues/2511

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/ast/Symbol.scala [hunk 1]
```diff
@@ -312,6 +322,34 @@ object Symbol {
     override def toString: String = if (namespace.isEmpty) name else namespace.mkString("/") + "." + name
   }
 
+  /**
+    * Instance Symbol.
+    */
+  final class InstanceSym(val id: Int, val clazz: Symbol.ClassSym, val loc: SourceLocation) {
+    /**
+      *
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/Symbol.scala:330]: Woops

**Comment by magnus-madsen** [comment]
Is this a sacrifice on the alter of convenience or is it a good thing?

**Comment by magnus-madsen** [comment]
What is an instance symbol in a semantic sense?

**Comment by mlutze** [comment]
I think it makes sense to have some way to refer to a specific instance.

**Comment by magnus-madsen** [comment]
Alright, I am convinced.

**Comment by magnus-madsen** [comment]
We should also change this line:

https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/api/lsp/provider/SemanticTokensProvider.scala#L143

I think we should construct a new semantic token for the location which is like of type `Class` or maybe `Interface`?

**Comment by mlutze** [comment]
> We should also change this line:
> 
> https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/api/lsp/provider/SemanticTokensProvider.scala#L143
> 
> I think we should construct a new semantic token for the location which is like of type `Class` or maybe `Interface`?

I think `type class` -> `interface` and `instance -> class` is the best correspondence.

**Comment by magnus-madsen** [comment]
> > We should also change this line:
> > https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/api/lsp/provider/SemanticTokensProvider.scala#L143
> > I think we should construct a new semantic token for the location which is like of type `Class` or maybe `Interface`?
> 
> I think `type class` -> `interface` and `instance -> class` is the best correspondence.

Agree and same thought.

**Comment by magnus-madsen** [comment]
> We should also change this line:
> 
> https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/api/lsp/provider/SemanticTokensProvider.scala#L143
> 
> I think we should construct a new semantic token for the location which is like of type `Class` or maybe `Interface`?

If you make this change, I can merge it in.

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**